### ISSUE TITLE
Feature request: Allow data.table syntax on a lazy_dt

### ISSUE THREAD

**Issue by pnacht** [tidyverse/dtplyr#134]
What's the feasibility of allowing lazy_dt's to use `[.data.table` syntax?

I do a lot of exploratory work where I want to run a quick operation on a table to check something on the console, and I find the data.table syntax much friendlier for this throw-away use-case (obviously dplyr wins by a mile in terms of legibility).

So my current workflow messing with dtplyr is to keep all my tables as data.tables, and then when I'm writing some operation I want to keep, I wrap it with `lazy_dt()` and `as.data.table()`:

```r
dt <- iris %>%
  lazy_dt() %>%
  filter(
    Species == "setosa"
  ) %>%
  as.data.table()
```

This is usually intentional. After all, if I later want to use this filtered `dt`, I might as well do the filtering only once by forcing it now, instead of accumulating operations every time I use it.

But there are other times where I wouldn't mind keeping tables as lazy_dt's. This is mostly tables which are rarely used alone, but which I subset in different ways prior to joining with other tables:

For example: 

``` r
suppressPackageStartupMessages({
  library(dplyr)
  library(dtplyr)
  library(data.table)
})

b1 <- data.table(beaver1)
b2 <- data.table(beaver2)
b1$day <- b1$day - 39  # to make the days match

b1 %>%
  lazy_dt() %>%
  left_join(
    b2 %>%
      lazy_dt() %>%
      select(
        day,
        time2 = time
      ),
    by = "day"
  ) %>%
  as.data.table()
#>       day time2 time  temp activ
#>    1: 307   930  840 36.33     0
#>    2: 307   940  840 36.33     0
#>    3: 307   950  840 36.33     0
#>    4: 307  1000  840 36.33     0
#>    5: 307  1010  840 36.33     0
#>   ---                           
#> 8212: 308   120  340 37.15     1
#> 8213: 308   130  340 37.15     1
#> 8214: 308   140  340 37.15     1
#> 8215: 308   150  340 37.15     1
#> 8216: 308   200  340 37.15     1
```

<sup>Created on 2019-12-04 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

Let's say I don't often need to check the set `b2`, but I often join it with other tables, each time performing a different `select()` operation on it.

I currently need to do `b2 %>% lazy_dt() %>% select(...)` each time, since omitting the `lazy_dt()` call throws a warning.

Now, since I don't often inspect `b2` directly, I could simply define it as a lazy_dt(). However, this then makes it annoying to perform those throw-away console inspections on it from time to time, since the `[.data.table` syntax isn't available.

This is an infrequent demand, so certainly low-priority, but if exposing the data.table syntax isn't complex, it'd certainly be appreciated.

**Comment by hadley** [comment]
I think this is currently out of scope for dtplyr, because of the challenge of understanding the mixed eager/lazy semantics of what you want.
